Migrate to TypeScript 6 (tsconfig types field + ecosystem compat check)

## Problem

Dependabot opened #39 to bump TypeScript `5.9.3 → 6.0.2`. The build fails on every Node matrix version with TS2591 errors:

```
src/index.ts(3,31): error TS2591: Cannot find name 'node:module'. Do you need to install type definitions for node?
src/index.ts(49,3): error TS2591: Cannot find name 'process'. Do you need to install type definitions for node?
src/lib/api-client.ts(106,20): error TS2591: Cannot find name 'process'. Do you need to install type definitions for node?
src/lib/api-client.ts(107,21): error TS2591: Cannot find name 'process'. Do you need to install type definitions for node?
```

## Root cause

TypeScript 6 changed its default behaviour for `@types/*` package auto-discovery. Under 5.x, `@types/node` was implicitly included when present in `node_modules/@types`; under 6.x, it is not, unless the compiler is told to include it explicitly via the `types` field in `tsconfig.json`.

Current `tsconfig.json` has no `types` entry:

```json
{
  "compilerOptions": {
    "target": "ES2022",
    "module": "Node16",
    "moduleResolution": "Node16",
    "outDir": "./build",
    "rootDir": "./src",
    "strict": true,
    ...
  }
}
```

So all the `process.*` and `node:*` imports in `src/` resolve under 5.x but not 6.x.

## Migration checklist

- [ ] Add `"types": ["node"]` to `compilerOptions` in `tsconfig.json`.
- [ ] Verify the build passes on all three Node matrix versions (18, 20, 22).
- [ ] Scan for any other breaking changes from the TS 6 release notes that might bite us (there are a handful — stricter narrowing, changed lib.d.ts internals, etc.). Release notes: https://devblogs.microsoft.com/typescript/ (filter for 6.0).
- [ ] Confirm `@modelcontextprotocol/sdk`, `zod`, and `typescript-eslint` are all compatible with TS 6. At time of writing, `typescript-eslint` explicitly supports TS ≥ 5.4; TS 6 support may lag.
- [ ] Bump `typescript` and `@typescript-eslint/*` in one coordinated PR.

## Why a dedicated PR

TS 6 is a major version bump with real semantic changes — not safe to rubber-stamp a Dependabot auto-merge. It should ship as its own feature-branch PR where the diff is reviewable and we can bundle whatever `typescript-eslint` / SDK changes become necessary in the same change.

## Context

Discovered while triaging open Dependabot PRs after #89 merged. Closed #39 (Dependabot's auto-bump attempt) in favour of this tracked migration.

--- a/package.json
+++ b/package.json
@@ -84,7 +84,7 @@
     "lint-staged": "^16.4.0",
     "prettier": "^3.8.3",
     "tsx": "^4.19.0",
-    "typescript": "^5.7.0",
+    "typescript": "^6.0.3",
     "typescript-eslint": "^8.58.2",
     "vitest": "^3.2.4"
   },

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -3,6 +3,7 @@
     "target": "ES2022",
     "module": "Node16",
     "moduleResolution": "Node16",
+    "types": ["node"],
     "outDir": "./build",
     "rootDir": "./src",
     "strict": true,

